File: 136.single-number.py
#
# @lc app=leetcode id=136 lang=python3
#
# [136] Single Number
#

# @lc code=start
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        xor = 0

        for num in nums:
            xor ^= num
        
        return xor
        
# @lc code=end

File: 136.single-number.java
/*
 * @lc app=leetcode id=136 lang=java
 *
 * [136] Single Number
 */

// @lc code=start

import java.util.Set;

class Solution {
    public int singleNumber(int[] nums) {
        int xor = 0;

        for(int num : nums) {
            xor = xor ^ num;
        }

        return xor;
        
    }
}
// @lc code=end

